platform-grpc-service-framework/src/main/java/org/hypertrace/core/serviceframework/grpc/GrpcPlatformServiceContainer.java

```diff
@@ -199,6 +209,20 @@ public boolean healthCheck() {
     }
   }
 
+  private Duration[] generateLatencyHistogramBuckets() {
+    ArrayList<Duration> histogram_buckets = new ArrayList<>();
+    Duration curr_duration =
+        getAppConfig().hasPath(MAX_LATENCY_HISTOGRAM_BUCKET_DURATION)
+            ? getAppConfig().getDuration(MAX_LATENCY_HISTOGRAM_BUCKET_DURATION)
+            : DEFAULT_MAX_LATENCY_HISTOGRAM_BUCKET_DURATION;
+    while (curr_duration.compareTo(DEFAULT_MIN_LATENCY_HISTOGRAM_BUCKET_DURATION) > 0) {
+      histogram_buckets.add(curr_duration);
+      curr_duration = curr_duration.dividedBy(LATENCY_HISTOGRAM_BUCKET_FACTOR);
+      curr_duration = Duration.ofMillis(curr_duration.toMillis()); // Round off to milliseconds
+    }
+    return histogram_buckets.toArray(new Duration[histogram_buckets.size()]);
```

Comment:
histogram_buckets will be descending order? is that okay with micrometer spec?

Reply:
I don't think we have any problem with that. I tested it on dev-sandbox. Was able to see the metrics being generated on dev-sandbox for license-metering-service and actor-service.